lib/grandpa/types.go

```diff
@@ -62,6 +94,29 @@ func NewVoteFromHeader(h *types.Header) *Vote {
 	}
 }
 
+// NewVoteFromHash returns a new Vote given a hash and a blockState
+func NewVoteFromHash(hash common.Hash, blockState BlockState) (*Vote, error) {
+	has, err := blockState.HasHeader(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	if !has {
```

Comment:
this could be a else if, since its related to previous statement.

Reply:
same as above